src/UI/Components/Diagram/Context/ComposerEditorProvider.test.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { act } from "react";
+import { Route, Routes, useLocation } from "react-router-dom";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { StoreProvider } from "easy-peasy";
+import { HttpResponse, http } from "msw";
+import { setupServer } from "msw/node";
+import { RemoteData } from "@/Core";
+import { getStoreInstance } from "@/Data";
+import { dependencies } from "@/Test";
+import {
+  DependencyProvider,
+  EnvironmentHandlerImpl,
+  PrimaryRouteManager,
+} from "@/UI";
+import CustomRouter from "@/UI/Routing/CustomRouter";
+import history from "@/UI/Routing/history";
+import "@testing-library/jest-dom";
+import {
+  childModel,
+  containerModel,
+  mockedInstanceWithRelations,
+  parentModel,
+} from "../Mocks";
+import { defineObjectsForJointJS } from "../testSetup";
+import { ComposerEditorProvider } from "./ComposerEditorProvider";
+
+const setup = (editable: boolean = false, instanceId: string) => {
```

Comment:
general habit, if you pass a default value, then that argument should go after the ones without defaults. There's no point in having a default otherwise.